### DISCUSSION TITLE
Handle prebuildify and node-pre-gyp, don't invoke prebuild-install for packages that don't use `prebuild-install`

### DISCUSSION THREAD

**Discussion by segevfiner** [yao-pkg/pkg#80] [Q&A]
It would be nice if pkg handled packages using `prebuildify`, `node-pre-gyp` and `prebuild-install` better:
1. It should only invoke `prebuild-install` or `node-pre-gyp` if actually supported and used by the package or there is custom configuration where custom prebuilt binaries have been built for a package that doesn't otherwise have them, otherwise it will just error, output a warning/error for unhandled `.node` files that are for the wrong platform. (For unsupported packages, users can place the right `.node` files themselves ATM, so it can still work even if unsupported directly by `pkg`). 
2. It should handle `prebuildify` and package only the binaries for the right platform, it currently packages all of them, which can increate binary size significantly, and there is not per target ignore mechanism to workaround it currently. It should also try to correctly package needed additional vendored DLLs that are used by the `.node` module, e.g. are alongside it, and only the ones for the correct platform.
3. It should handle `node-pre-gyp` packages.

**Comment by robertsLando** [comment]
PR?

**Reply by segevfiner** [reply]
I will, if I will have time, and won't forget. 😉
